packages/api/src/eventListener.ts

```diff
@@ -750,6 +756,14 @@ export default class EventListener extends IndexPriceInterface {
 		fMarkPricePremium: BigNumber,
 		fSpotIndexPrice: BigNumber,
 	): void {
+		// Funding rate will be zero before initialization. It should be rare,
+		// but it is possible to receive an event while eventListener is
+		// initializing.
+		if (!this.isInitialized) {
+			console.log("onUpdateMarkPrice: eventListener not initialized");
+			return;
+		}
+
```

Comment:
if this is possible, then it is also possible that we receive the event between assigning this.isInitialized=true and calling this._update

Reply:
this is now changed, we set isInitialized only after first update call